homeassistant/components/homeassistant/scene.py

```diff
@@ -121,9 +127,9 @@ def _ensure_no_intersection(value):
 class SceneConfig(NamedTuple):
     """Object for storing scene config."""
 
-    id: str
+    id: str | None
     name: str
-    icon: str
+    icon: str | None
```

Comment:
I have had to mark `id` and `icon` as optional due to this initialisation code.
https://github.com/home-assistant/core/blob/41c497ee6e4d32e3316af653f5f03b3ba679eb9c/homeassistant/components/homeassistant/scene.py#L246

Reply:
Yes, also that fits with 
https://github.com/home-assistant/core/blob/41c497ee6e4d32e3316af653f5f03b3ba679eb9c/homeassistant/components/homeassistant/scene.py#L269-L273
which (by using `get()`) seems to imply that `id` and `icon` are allowed to be undefined/None.

Reply:
Yes, that also fits with
https://github.com/home-assistant/core/blob/41c497ee6e4d32e3316af653f5f03b3ba679eb9c/homeassistant/components/homeassistant/scene.py#L269-L273
Which implies that `id` and `icon` are allowed to be undefined/None.